riposte-core/src/test/java/com/nike/riposte/util/AsyncNettyHelperTest.java

```diff
@@ -857,4 +866,84 @@ public void executeOnlyIfChannelIsActive_sets_original_tracing_and_mdc_info_back
         );
         assertThat(postCallTracingInfo).isEqualTo(preCallTracingInfo);
     }
+
+    @Test
+    public void executeAsyncCall_shouldReturnCompletableFuture() throws Exception {
```

Comment:
All these tests should have their completable futures capture the current tracing state at the time of execution and then verify that it matches the expected tracing state (which you're setting up in `mockChannelHandlerContextWithTraceInfo()`). You're verifying span name for the subspans which is good, but you should also verify that the trace ID of the subspan is the expected trace ID and that the parent span ID matches the span ID of the parent. IOW verify that the subspan really is a true subspan.

Reply:
i've added those, and only used the 'withTraceInfo' on those specific tests.